lib/px/service/kinesis/base_request.rb

```diff
@@ -33,15 +33,67 @@ def initialize
       @last_send = Time.now
       @last_throughput_exceeded = nil
 
-      @buffer = Array.new
+      @buffer = []
     end
 
     ##
     # Check if buffer should be flushed and sent to kinesis
-    #
     def flush_records
-      if (put_rate_decay == DEFAULT_PUT_RATE && @buffer.length >= FLUSH_LENGTH) || (Time.now - @last_send > put_rate_decay)
-        response = @kinesis.put_records(:stream_name => @stream, :records => @buffer)
+      begin
+        _flush_records
```

Comment:
Need to wrap the circuit method in this to translate random exceptions into our standard ones.